pkg/asset/installconfig/aws/platform.go

```diff
@@ -18,6 +20,24 @@ func Platform() (*aws.Platform, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get AWS public regions: %w", err)
 	}
+	longRegions := make([]string, 0, len(aws.RegionLookupMap))
+	shortRegions := make([]string, 0, len(aws.RegionLookupMap))
+	for _, region := range regions {
+		if longName, ok := aws.RegionLookupMap[region]; ok {
+			longRegions = append(longRegions, fmt.Sprintf("%s (%s)", region, longName))
+			shortRegions = append(shortRegions, region)
```

Comment:
Should we append the region-code even if the long name is not available?

I think the returned `regions` should be the right source of available regions, not the `RegionLookupMap`, right?

Reply:
Hmm, it's quite tough as we are using the `longRegions` as options... 

Maybe, for region that doesn't have a long name, we set its long name to the region code (short name) itself? The survey will look like:

```bash
  sa-east-1 (South America (São Paulo))
  sa-new-region
> us-east-1 (US East (N. Virginia))
  us-new-region
```

Reply:
I don't mind making that change. I believe the original way was to limit to what we know. I also think GCP does this.

Reply:
@patrickdillon Do you have a preference ?

Reply:
```
  me-south-1 (Middle East (Bahrain))
  mx-central-1 (Mexico (Central))
  sa-east-1 (South America (São Paulo))
> test-region-1
  us-east-1 (US East (N. Virginia))
  us-east-2 (US East (Ohio))
  us-west-1 (US West (N. California))
```
Here is the example from the code. I added the test-region to the list of returned regions through the api calls. This shows that when a region that we do not know about is returned it will still be shown (without a long name).

Reply:
Cool, I was able to produce the same behaviour too.

```bash
? Region  [Use arrows to move, type to filter, ? for more help]
  me-south-1 (Middle East (Bahrain))
  mx-central-1 (Mexico (Central))
  sa-east-1 (South America (São Paulo))
> unknown-us-region
  us-east-1 (US East (N. Virginia))
  us-east-2 (US East (Ohio))
  us-west-1 (US West (N. California))
```